scala/src/main/scala/smile/validation/Normalize_Mutual_Index.scala

```diff
@@ -0,0 +1,67 @@
+import scala.math.{max, log, sqrt}
+
+/**
+ * @author Beck Gaël
+ */
+object Externals_Indexes
+{
+	/**
+	 * Normalize Sequences in order to prevent construction of a to big 'count' matrix
+	 * Ex: [4,5,6,6] -> [0,1,2,2]
+	 **/
+	private def prepareList(x: Array[Int]) =
+	{
+		val indexedValuesMap = x.distinct.zipWithIndex.toMap
+		x.map(indexedValuesMap)
+	}
+
+	private def mutualInformationInteral(x: Array[Int], y:Array[Int]) =
+	{
+		require( x.size == y.size )
```

Comment:
The implementation is overall imperative rather than functional. So why not we implement it in Java so that Java users can use it too? Again, for fundamental things, I prefer Java for bigger audience. For high algorithms, I am fine with Scala implementation as long as the interface can be called from Java too.

Sorry for miscommunication in earlier gitter chat.